util/config_stripper.py

```diff
@@ -78,11 +96,61 @@ def strip_tar(input, output):
     shutil.rmtree(tempdir)
     return 0
 
-
-def strip_config(path):
+def strip_layer(path):
+    # The original layer tar is of the form <random string>/layer.tar, the
+    # working directory is one level up from where layer.tar is.
+    original_dir = os.path.normpath(os.path.join(os.path.dirname(path), '..'))
+
+    buf = cStringIO.StringIO()
+
+    # Go through each file/dir in the layer
+    # Set its mtime to 0
+    # If it's a file, add its content to the running buffer
+    # Add it to the new gzip'd tar.
+    with tarfile.open(name=path, mode='r') as it:
+      with tarfile.open(fileobj=buf, mode='w') as ot:
+        for tarinfo in it:
+          tarinfo.mtime = 0
+          if tarinfo.isfile():
+            f = it.extractfile(tarinfo)
+            ot.addfile(tarinfo, f)
+          else:
+            ot.addfile(tarinfo)
+
+    # Create the new diff_id for the config
+    tar = buf.getvalue()
+    diffid = hashlib.sha256(tar).hexdigest()
+    diffid = 'sha256:%s' % diffid
+
+    # Compress buf to gz
+    # Shelling out to bash gzip is noticeably faster than using python's gzip
+    gzip_process = subprocess.Popen(
+        ['gzip', '-nf'],
+        stdout=subprocess.PIPE,
+        stdin=subprocess.PIPE,
+        stderr=subprocess.PIPE)
+    gz = gzip_process.communicate(input=tar)[0]
+
+    # Calculate sha of layer
+    sha = hashlib.sha256(gz).hexdigest()
+    new_name = 'sha256:%s' % sha
+    with open(os.path.join(original_dir, new_name), 'w') as out:
+      out.write(gz)
+
+    shutil.rmtree(os.path.dirname(path))
+    return (new_name, diffid)
+
+
+def strip_config(path, new_diff_ids):
     with open(path, 'r') as f:
         config = json.load(f)
     config['created'] = _TIMESTAMP
+    config['rootfs']['diff_ids'] = new_diff_ids
+
+    # Base container info is not required and changes every build, so delete it.
+    del config['container']
```

Comment:
I think these all need to be inside if statements so you only delete them if they exist:
```shell
    del config['container']
KeyError: 'container'
```

The bazel rules generate images without these.